ui/job-view/pushes/FuzzyJobFinder.jsx

```diff
@@ -0,0 +1,324 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+  Button,
+  Col,
+  FormGroup,
+  Label,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  InputGroup,
+  Input,
+} from 'reactstrap';
+import Fuse from 'fuse.js';
+
+import PushModel from '../../models/push';
+import { withNotifications } from '../../shared/context/Notifications';
+import { formatTaskclusterError } from '../../helpers/errorMessage';
+import { sortAlphaNum } from '../../helpers/sort';
+
+class FuzzyJobFinder extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      fuzzySearch: '',
+      fuzzyList: [],
+      selectedList: [],
+      removeDisabled: true,
+      addDisabled: true,
+      submitDisabled: false,
+    };
+  }
+
+  /*
+   *  Filter the list of runnable jobs based on the value of this input.
+   *  Only actually do the filtering when `enter` is pressed, as filtering 13K DOM elements is slow...
+   *  If this input is empty when `enter` is pressed, reset back to the full list of runnable jobs.
+   */
+  filterJobs = ev => {
+    // By default we show a trimmed down list of runnable jobs, but there's an option to show the full list
+    let currentList;
+    if (this.state.useFullList) {
+      currentList = this.props.jobList;
+    } else {
+      currentList = this.props.filteredJobList;
+    }
+
+    if (ev && ev.type === 'keydown') {
+      if (ev.keyCode === 13) {
```

Comment:
Use of ``keyCode`` is now deprecated.  We should use ``key`` instead: https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent

Reply:
Done. Looks like there's a couple other uses of keyCode in the repo that should get fixed at some point: https://github.com/mozilla/treeherder/search?q=keyCode&unscoped_q=keyCode

Reply:
Oh, heh.  I guess I missed those when I converted it.  My bad!  Travis has been a little wonky today.  But I'll keep an eye on this to see when the tests pass...